internal/controller/helmchart_controller.go

```diff
@@ -1359,15 +1367,78 @@ func (r *HelmChartReconciler) makeVerifiers(ctx context.Context, obj *helmv1.Hel
 				SubjectRegExp: match.Subject,
 			})
 		}
-		defaultCosignOciOpts = append(defaultCosignOciOpts, soci.WithIdentities(identities))
+		defaultCosignOciOpts = append(defaultCosignOciOpts, scosign.WithIdentities(identities))
 
-		verifier, err := soci.NewCosignVerifier(ctx, defaultCosignOciOpts...)
+		verifier, err := scosign.NewCosignVerifier(ctx, defaultCosignOciOpts...)
 		if err != nil {
 			return nil, err
 		}
 		verifiers = append(verifiers, verifier)
 		return verifiers, nil
+	case "notation":
+		// get the public keys from the given secret
+		secretRef := obj.Spec.Verify.SecretRef
+
+		if secretRef == nil {
+			return nil, fmt.Errorf("verification secret cannot be empty: '%s'", obj.Name)
+		}
+
+		verifySecret := types.NamespacedName{
+			Namespace: obj.Namespace,
+			Name:      secretRef.Name,
+		}
+
+		pubSecret, err := r.retrieveSecret(ctx, verifySecret)
+		if err != nil {
+			return nil, err
+		}
+
+		data, ok := pubSecret.Data[notation.DefaultTrustPolicyKey]
+		if !ok {
+			return nil, fmt.Errorf("'%s' not found in secret '%s'", notation.DefaultTrustPolicyKey, verifySecret.String())
+		}
+
+		var doc trustpolicy.Document
+
+		if err := json.Unmarshal(data, &doc); err != nil {
+			return nil, err
+		}
+
+		trustPolicy := notation.CleanTrustPolicy(&doc, ctrl.LoggerFrom(ctx))
+		defaultNotaryOciOpts := []notation.Options{
+			notation.WithTrustStore(trustPolicy),
+			notation.WithRemoteOptions(verifyOpts...),
+			notation.WithAuth(clientOpts.Authenticator),
+			notation.WithKeychain(clientOpts.Keychain),
+			notation.WithInsecureRegistry(clientOpts.Insecure),
+			notation.WithLogger(ctrl.LoggerFrom(ctx)),
+		}
+
+		for k, data := range pubSecret.Data {
+			if strings.HasSuffix(k, ".crt") || strings.HasSuffix(k, ".pem") {
+				verifier, err := notation.NewNotationVerifier(append(
+					defaultNotaryOciOpts,
+					notation.WithRootCertificate(data))...)
+				if err != nil {
+					return nil, err
+				}
+				verifiers = append(verifiers, verifier)
+			}
+		}
+		return verifiers, nil
```

Comment:
I've refactored things a bit in [5e2074a](https://github.com/fluxcd/source-controller/pull/1075/commits/5e2074a49a799169ca60f3f07651a71ba9798f52). I've removed the need of a loop so that `verifiers` should not be empty (assuming no errors occurred which would have already returned before getting to the point `verifiers` is returned). I can add a check like how cosign handles things but it's a bit redundant.